lib/webdriverio.js

```diff
@@ -230,6 +234,14 @@ let WebdriverIO = function (args, modifier) {
             })
     }
 
+    function createRequestHanlderDecorator (opts) {
```

Comment:
No, if you look more attentively you will see that it is decorate only screenshot request, because in takeScreenshot method created a new client instance.
Moreover I've checked it a lot of times and I can say with confidence that it decorates only screenshot request  on fail.